### PR TITLE
Doc update with mlg extension - fix #10855

### LINKED ISSUE #10855: Comments and doc still mention .ml4 even though things are now .mlg
#### Description of the problem

At some point recently, we replaced `ml4` with `.mlg`, but the documentation was not fully updated.
```
$ git grep '\.ml4'
.gitignore:# .ml4 / .mlp files
dev/doc/archive/extensions.txt:  type et du bon nom dans le module Tactic de pcoq.ml4
dev/doc/archive/extensions.txt:- il faut aussi exporter la règle hors de g_tactic.ml4. Pour cela, il
dev/doc/build-system.dev.txt:  - Btw, .ml4 are explicitly turned into .ml files, which stay after build.
dev/doc/build-system.dev.txt:    restricted set of .ml4 (see variable BUILDGRAMMAR).
dev/doc/build-system.dev.txt:.ml4 files
dev/doc/build-system.dev.txt:.ml4 are converted to .ml by camlp5. By default, they are produced
dev/doc/build-system.dev.txt: - no risk of editing them by mistake instead of the .ml4
dev/doc/build-system.dev.txt: - the location in the binary .ml are those of the initial .ml4,
dev/doc/build-system.dev.txt:   hence errors are properly reported in the .ml4.
dev/doc/build-system.txt:q_constr.cmo), then see sections ".ml4 files" and "new files".
dev/doc/build-system.txt:.ml4/.mlp files
dev/doc/build-system.txt: - a .ml4 is now always preprocessed with grammar.cma (and q_constr.cmo),
dev/doc/build-system.txt:   except coqide_main.ml4 and its specific rule
dev/doc/changes.md:- Support for `.ml4` files, processed by camlp5 has been removed in
dev/doc/changes.md:  `file.ml4` file to `file.mlg` and adding a few brackets.
dev/doc/changes.md:extension of a macro file from `.ml4` to `.mlg`. Not all camlp5 macros are
dev/doc/changes.md:Files plugins/*/g_*.ml4 take the place of files plugins/*/*.v.
dev/doc/changes.md:Tactics depending on Init now tactics/*.ml4 (no longer in tactics/*.v)
dev/doc/coq-src-description.txt: to pre-process .ml4 files containing EXTEND constructions,
dev/v8-syntax/syntax-v8.tex:%% eqdecide.ml4
doc/sphinx/practical-tools/utilities.rst:    src/baz.ml4
doc/sphinx/practical-tools/utilities.rst:  extensions (like ``VERNAC EXTEND``) must use the ``.ml4`` extension
lib/coqProject_file.ml:        | ".ml4" ->
parsing/pcoq.ml:    keywords declared in g_*.ml4 *)
parsing/pcoq.mli:        | macro-generation in tacextend.ml4/vernacextend.ml4/argextend.ml4
plugins/cc/README:- cctac.ml4 : the tactic itself
plugins/cc/ccproof.ml:(* proof-trees that will be transformed into proof-terms in cctac.ml4   *)
plugins/ltac/g_obligations.mlg:  Elaborated from correctness/psyntax.ml4 by Jean-Christophe Filliâtre *)
tools/CoqMakefile.in:# a .ml4 file
tools/coqdoc/tokens.ml:    (code copied from parsing/lexer.ml4 for the use of coqdoc, Apr 2010)
```

Some of these are probably fine (such as the comment in `tools/coqdoc/tokens.ml`), while others are clearly not (such as the documentation in `doc/sphinx/practical-tools/utilities.rst`).

#### Coq Version

e01c9132fc505a3c443bee27d88dcd16a566616f

### PR DESCRIPTION
**Pull request by mcaci**
**Kind:** documentation.

Fixes / closes #10855

### CODE REVIEW

**Review by gares — APPROVED** [review]

File: dev/doc/archive/extensions.txt [hunk 1]
```diff
@@ -10,9 +10,9 @@ Exemple de l'ajout de l'entrée "clause":
   ce nouveau ClauseArgType
 
 - utiliser le rawwit_clause pour définir une entrée clause du bon
-  type et du bon nom dans le module Tactic de pcoq.ml4
+  type et du bon nom dans le module Tactic de pcoq.mlg
```
Comment (JasonGross) [inline dev/doc/archive/extensions.txt]: I think probably the .ml4 in archive.txt should stay as they are, as they are historical references, not meant to describe the current system.

File: dev/doc/build-system.dev.txt [hunk 2]
```diff
@@ -69,16 +69,16 @@ This file documents internals of the implementation of the make-based
 build system. For what a Coq developer needs to know about the build
 system, see build-system.txt and build-system.dune.md .
 
-.ml4 files
+.mlg files
 ----------
 
-.ml4 are converted to .ml by camlp5. By default, they are produced
+.mlg are converted to .ml by camlp5. By default, they are produced
 in the binary ast format understood by ocamlc/ocamlopt/ocamldep.
```
Comment (JasonGross) [inline dev/doc/build-system.dev.txt]: @ppedrot, this section is wrong now, right?  The .mlg are turned into .ml by coqpp, not camlp5, they are not in binary format, and the locations are only approximately right?

File: dev/doc/changes.md [hunk 3]
```diff
@@ -42,11 +42,11 @@ Proof state:
 
 ### ML4 Pre Processing
 
-- Support for `.ml4` files, processed by camlp5 has been removed in
+- Support for `.mlg` files, processed by camlp5 has been removed in
```
Comment (JasonGross) [inline dev/doc/changes.md]: changes.md should not be changed.  This sentence in particular becomes nonsensical with the change

**Review by JasonGross — CHANGES_REQUESTED** [review]

**Comment by mcaci** [comment]
Hi @JasonGross I have reverted all changes for the aformentioned files for the moment. Are the other files ok with the changes?

File: dev/doc/build-system.dev.txt (hunk 2 above)
Comment (ppedrot) [reply dev/doc/build-system.dev.txt]: Yes to all.

File: dev/doc/build-system.txt [hunk 4]
```diff
@@ -89,7 +89,7 @@ enables partial recalculation of dependencies (only the dependencies
 of changed files are recomputed).
 
 If you add a dependency to a Coq camlp5 extension (grammar.cma or
-q_constr.cmo), then see sections ".ml4 files" and "new files".
+q_constr.cmo), then see sections ".mlg files" and "new files".
```
Comment (ppedrot) [inline dev/doc/build-system.txt]: This documentation file is in dire need of an update. Lots of comments are fairly outdated, including this one.

File: plugins/ltac/g_obligations.mlg [hunk 5]
```diff
@@ -10,7 +10,7 @@
 
 (*
   Syntax for the subtac terms and types.
-  Elaborated from correctness/psyntax.ml4 by Jean-Christophe Filliâtre *)
+  Elaborated from correctness/psyntax.mlg by Jean-Christophe Filliâtre *)
```
Comment (JasonGross) [inline plugins/ltac/g_obligations.mlg]: Oops, I missed this one before.  I suspect this one is also historical, and thus better left alone.

**Review by JasonGross — APPROVED** [review]
LGTM (though I think the one in plugins/ltac/g_obligations.mlg is better reverted)

File: dev/doc/build-system.txt (hunk 4 above)
Comment (JasonGross) [reply dev/doc/build-system.txt]: I think this can be left for another PR, though

File: plugins/ltac/g_obligations.mlg (hunk 5 above)
Comment (mcaci) [reply plugins/ltac/g_obligations.mlg]: No problem :) I'm also reverting this one

**Comment by mcaci** [comment]
The list of files not modified in the PR at the moment is
```sh
$ git grep --name-only '\.ml4' 
dev/doc/archive/extensions.txt
dev/doc/build-system.dev.txt
dev/doc/changes.md
lib/coqProject_file.ml
plugins/ltac/g_obligations.mlg
tools/coqdoc/tokens.ml
```

**Comment by mcaci** [comment]
I'm going to check later in the day what causes the failure in ci-perennial build. The difference with the last successful build is that I restored to the old version the file `plugins/ltac/g_obligations.mlg`

**Comment by SkySkimmer** [comment]
It's a spurious failure, don't worry about it

Gaëtan Gilbert

**Comment by Zimmi48** [comment]
Can any of the existing reviewers self-assign? I don't think any additional review is needed now.

**Pull request merged**